src/main/java/com/newlinegaming/Runix/SigBlock.java

```diff
@@ -2,21 +2,34 @@
 
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
 
 /**
  * Plain old Data.  getClass() and meta.  Use this to preserve all your block info.
  * Meta is the additional information like orientation or color used in some blocks.
  * Doesn't currently do TileEntity or NBT.
  */
+
+//TODO: only use block state
 @SuppressWarnings("unused")
-public class SigBlock{
-    public final int meta;
-    public final Block blockID;
-    public SigBlock(Block blockID, int meta){
-        this.blockID = blockID;
-        this.meta = meta;
+public class SigBlock {
+
+    private IBlockState state;
+    private Block block;
```

Comment:
I can see how doing this is correct for running.  However, these values or a proxy need to be capable of being pickled in guava for the Runix save state json.  Have you tested things like Rubric still work with this change?

Reply:
Its been so long I cant remember if Rubric ever worked might be thinking of a different rune...right now I care about getting it running and I can we might need to refine the format for the json(I think we should move to forges capabilities system if possible but that's nbt but its also smaller in the long run) but using metadata as we knew is gone and everything relies on a blockstate we could hack around that and give it the state name instead of feeding it the full blockstate... actully we could get rid of Block

Reply:
I think you're right.  There's not too many runes that use Block json saves right now.  What's not going to go away is runes referencing the material of a block to see if it is (in general) the same as another block.  I suspect the metadataWhitelist will get more complicated now.